src/async-wrap.cc

```diff
@@ -296,6 +296,14 @@ static void PromiseHook(PromiseHookType type, Local<Promise> promise,
   PromiseWrap* wrap = Unwrap<PromiseWrap>(promise);
   if (type == PromiseHookType::kInit || wrap == nullptr) {
     bool silent = type != PromiseHookType::kInit;
+    // set parent promise's async Id as this promise's triggerId
+    if (parent->IsPromise()) {
+      // parent promise exists, current promise
+      // is a chained promise, so we set parent promise's id as
+      // current promise's triggerId
+      double trigger_id = Unwrap<PromiseWrap>(parent.As<Promise>())->get_id();
```

Comment:
@addaleax , thank you, so it will look like

```
auto parent_wrap = Unwrap<PromiseWrap>(parent.As<Promise>());
if (parent_wrap == nullptr) {
  // create a new PromiseWrap for parent promise with silent parameter
  parent_wrap = new PromiseWrap(env, parent, true);  
  parent_wrap->MakeWeak(parent_wrap);
}
// get id from parentWrap
double trigger_id = parent_wrap->get_id();
env->set_init_trigger_id(trigger_id);
```